pydm/tests/widgets/test_pushbutton.py

```diff
@@ -0,0 +1,566 @@
+            else:
+                assert pydm_pushbutton.confirmMessage == PyDMPushButton.DEFAULT_CONFIRM_MESSAGE
+
+        if not is_value_relative or channel_type == str:
+            # Limit the decimal places since Python 2.7. provided more floating point
+            assert signals.value == pydm_pushbutton.channeltype(pydm_pushbutton.pressValue)
+        else:
+            assert signals.value == updated_value + pydm_pushbutton.channeltype(pydm_pushbutton.pressValue)
+
+
+@pytest.mark.parametrize("current_channel_value, updated_value", [
+    # Current channel value type is array, getting a new int value
+    (np.array([123, 456]), 10),
+
+    # Test if the current channel value type is int, and the widget is getting new int, float, or string value
+    (10, 20),
+    (10, 20.20),
+    (10, "100"),
+
+    # Test if the current channel value type is float, and the widget getting new int, float, or string value
+    (10.10, 20.20),
+    (10.10, 42),
+    (10.10, "100.5"),
+
+    # Test if the current channel value type is string, and the widget is getting new int, float, or string value
+    ("Old str value", "New str value"),
+    ("Old str value", 42),
+    ("Old str value", 10.10),
+])
+def test_update_press_value(qtbot, signals, current_channel_value, updated_value):
+    """
+    Test the conversion of a new press value given the existing channel type.
+    Expectations:
+    For supported types (int, float, str), the conversions to the existing value type must be successful.
+
+    Parameters
+    ----------
+    qtbot : fixture
+        Window for widget testing
+    signals : fixture
+        The signals fixture, which provides access signals to be bound to the appropriate slots
+    current_channel_value : int, float, str, ndarray
+        The existing channel value, which will remain consistent with the type of the updated value, i.e. it must be
+        possible to convert the updated value to the type of the existing channel value
+    updated_value : int, float, str
+        The new channel to send to the channel by clicking on the widget
+    """
+    pydm_pushbutton = PyDMPushButton()
+    qtbot.addWidget(pydm_pushbutton)
+
+    # First, set the current channel type
+    signals.new_value_signal[type(current_channel_value)].connect(pydm_pushbutton.channelValueChanged)
+    signals.new_value_signal[type(current_channel_value)].emit(current_channel_value)
+
+    # Verify the new value can be converted/cast as long as the casting can be done
+    signals.new_value_signal[type(updated_value)].connect(pydm_pushbutton.updatePressValue)
+    signals.new_value_signal[type(updated_value)].emit(updated_value)
+
+    # Verify the new value is assigned to be the new pressValue as a str
+    assert pydm_pushbutton.pressValue == str(type(current_channel_value)(updated_value))
+
+def test_password():
+    pass
+
+
+@pytest.mark.parametrize("alarm_severity, alarm_sensitive_content", [
```

Comment:
Please also add a case in which you change the `alarm_sensitive_border` and make sure that it does not affect the Button stylesheet... for the PyDMPushButton just the alarm_sensitive_content matters.
Please add a True and False condition for the property.